### PR TITLE
[OBMIN-397] ellipsis text

### PR DESCRIPTION
**Pull request by Voloshkina**
Created ellipsis text and replaced in SB components

### CODE REVIEW

File: src/components/ellipsis-text/index.tsx [hunk 1]
```diff
@@ -1,28 +1,25 @@
 import { useRef, useEffect, useState } from 'react';
-import { Tooltip } from 'react-tooltip';
+import { useDelayTooltip } from 'hooks/useDelayTooltip';
```
Comment (Wolshebnik) [inline src/components/ellipsis-text/index.tsx:2]: move below

File: src/components/ellipsis-text/types.ts [hunk 2]
```diff
@@ -41,7 +41,19 @@ export type Cursor =
...
-export interface ITooltipProps extends ITooltip {
-  width?: number;
+export type Position = 'right' | 'left' | 'top' | 'bottom';
+
+export interface ITooltipProps {
+  gap?: number;
+  delay?: number;
   cursor?: Cursor;
+  position?: Position;
+  children: ReactNode;
```
Comment (Wolshebnik) [inline src/components/ellipsis-text/types.ts:51]: remove

File: src/components/ellipsis-text/index.tsx [hunk 3]
```diff
@@ -1,28 +1,25 @@
...
-  width = 200,
-  noArrow = true,
+  delay = 500,
   cursor = 'auto',
-  place = 'bottom',
-  delayHide = delay,
-  delayShow = delay,
-  className = 'custom-tooltip',
-  ...props
+  widthTooltip = 200,
```
Comment (Wolshebnik) [inline src/components/ellipsis-text/index.tsx:14]: try to change name

File: src/components/ellipsis-text/index.tsx [hunk 4]
```diff
@@ -40,33 +37,53 @@ export const EllipsisText = ({
...
-            {children}
-          </Tooltip>
-        </WrapTooltip>
+      {open && (
+        <Styles.Tooltip
+          gap={gap}
+          position={position}
+          widthEl={widthEl || 0}
+          heightEl={heightEl || 0}
+          widthTooltip={widthTooltip}
```
Comment (Wolshebnik) [inline src/components/ellipsis-text/index.tsx:80]: try to change name

File: src/components/ellipsis-text/types.ts [hunk 5]
```diff
@@ -1,4 +1,4 @@
-import { ITooltip } from 'react-tooltip';
+import { ReactNode } from 'react';
```
Comment (Wolshebnik) [inline src/components/ellipsis-text/types.ts:1]: remove

File: src/components/product-owner-data/index.tsx [hunk 6]
```diff
@@ -54,9 +51,9 @@ const ProductOwnerData = ({
             )}
 
             <Styles.PostReqDataText color="black" style={{ width: '200px' }}>
-              {/* <EllipsisText id={city} width={300}> */}
-              {city}
-              {/* </EllipsisText> */}
+              <EllipsisText gap={4} justifyContent={'start'} widthTooltip={300}>
```
Comment (Wolshebnik) [inline src/components/product-owner-data/index.tsx:54]: remove brackets

**Review by Wolshebnik — APPROVED** [review]

File: src/components/ellipsis-text/index.tsx [hunk 7]
```diff
@@ -40,33 +37,53 @@ export const EllipsisText = ({
     const heightRatio: number = scrollHeight / offsetHeight;
 
     if (widthRatio > 1.0 || heightRatio > 1.5) {
-      setToolTip(true);
+      setIsTooltipVisible(true);
     } else {
-      setToolTip(false);
+      setIsTooltipVisible(false);
+    }
+
```
Comment (cloudmenson) [inline src/components/ellipsis-text/index.tsx:44]: setIsTooltipVisible(widthRatio > 1.0 || heightRatio > 1.5 ? true : false);

File: src/hooks/useDelayTooltip.ts [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+      );
+    }
+
+    if (open) {
+      setTimer(null);
+      setPreviewOpen(true);
+      if (timer) {
+        clearTimeout(timer);
+      }
+    }
```
Comment (cloudmenson) [inline src/hooks/useDelayTooltip.ts:25]: if (open) {
  setTimer(null);
  setPreviewOpen(true);
  if (timer) clearTimeout(timer);
}

**Review by Kyslytsia — APPROVED** [review]

**Review by Viktoria-Serkiz — APPROVED** [review]

**Review by cloudmenson — APPROVED** [review]

**Review by ASGladchenko — APPROVED** [review]

**Pull request merged**